integration_tests/steps/two_sided_pay_as_clear.py

```diff
@@ -103,23 +104,23 @@ def test_offer_bid_market_clearing_rate_files(context):
 
 @then('one-on-one matching of offer & bid in PAC happens at bid rate')
 def one_on_one_matching_at_clearing_rate_at_bid_rate(context):
-    assert all(isclose(clearing[0], 30.0)
+    assert all(isclose(clearing, 30.0)
                for child in context.simulation.area.children
-               for market in child.past_markets
-               for clearing in list(market.state.clearing.values()))
+               for clearing in
+               list(child.bid_offer_matcher.match_algorithm.state.clearing.values()))
 
 
 @then('clearing rate is the bid rate of last matched bid')
 def clearing_rate_at_last_matched_bid_rate(context):
-    assert all(isclose(clearing[0], 15.0)
+    assert all(isclose(clearing, 15.0)
                for child in context.simulation.area.children
-               for market in child.past_markets
-               for clearing in list(market.state.clearing.values()))
+               for clearing in
+               list(child.bid_offer_matcher.match_algorithm.state.clearing.values()))
 
 
 @then('clearing rate is equal to the bid_rate')
 def clearing_rate_at_bid_rate(context):
-    assert all(isclose(clearing[0], 25.0)
+    assert all(isclose(clearing, 25.0)
                for child in context.simulation.area.children
-               for market in child.past_markets
-               for clearing in list(market.state.clearing.values()))
+               for clearing in
+               list(child.bid_offer_matcher.match_algorithm.state.clearing.values()))
```

Comment:
Nit: you can iterate over `.values()` without converting to list:
```suggestion
               child.bid_offer_matcher.match_algorithm.state.clearing.values())
```

Reply:
Yes sure, will adapt it